acstools/focus_diverse_epsfs.py

```diff
@@ -0,0 +1,349 @@
+and near the top of the WFC1 chip (and the detector overall). 
+
+>>>x = 2000
+>>>y = 4048
+>>>P = interp_epsf(ePSFs, x, y)
+
+Do the same as the previous example, but obtain the ePSF in detector space (instead of with 4x supersampling).
+>>>P = interp_epsf(ePSFs, x, y, pixel_space = True)
+
+Do the same as the previous example, but now in detector space and with subpixel offsets.
+>>>P = interp_epsf(ePSFs, x, y, pixel_space = True, subpixel_x = 0.77, subpixel_y = 0.33)
+
+
+More details for these examples are provided in a Jupyter notebook, which can be found at LINK. 
+
+
+"""
+
+import logging
+
+import dask
+import numpy as np
+import requests
+
+from dask.diagnostics import ProgressBar
+from requests.auth import HTTPBasicAuth
+from scipy import ndimage
+from urllib.request import urlopen, urlretrieve
+
+__taskname__ = "focus_diverse_epsfs"
+__author__   = "Gagandeep Anand, Yotam Cohen"
+__version__  = "1.0"
+__vdate__    = "02-Oct-2023"
+
+# Initialize the logger
+logging.basicConfig()
+LOG = logging.getLogger(f'{__taskname__}.Query')
+LOG.setLevel(logging.INFO)
+
+
+def psf_retriever(ipsoot, download_location):
+    """
+    Function to query API on AWS API Gateway for the ePSF FITS file that
+    corresponds to a given image rootname.
+
+    Returns
+    -------
+
+    Downloads the ePSF FITS file to download_location. The explicit return is 
+    the name of the downloaded FITS file.
+
+    """
+
+    # provide api URL and public key and ID
+    api_url = (
+        'https://8cclxcxse4.execute-api.us-east-1.amazonaws.com/main/psf-server-ops/')
+    api_id = 'iwx1prnqog'
+    api_key = 'T1fU5vycfM9KDlSLJoGBU6t0pzS0vjHKaSqmT6gU'
+
+    # stitch together credentials
+    auth = HTTPBasicAuth(api_id, api_key)
+
+    # send up post request with ipsoot event
+    myobj = {'ipsoot': ipsoot}
+    result = requests.post(api_url, json=myobj, auth=auth)
+
+    # grab url from result
+    url = result.text[1:-1]
+    remotefile = urlopen(url)
+
+    # determine readable name for file
+    content_disposition = remotefile.info()['Content-Disposition']
+    desired_filename = (content_disposition.split(';')[1].split('=')[1].split('-')[0] +
+                        '-' +
+                        content_disposition.split(';')[1].split('=')[1].split('-')[1].split('/')[-1])
+
+    # download file
+    urlretrieve(url, download_location + desired_filename)
+
+    return desired_filename
+
+
+def multi_psf_retriever(input_list, download_location, n_PROC=8, fromTextFile=False):
+    """
+    Function to batch query the API on AWS API Gateway for multiple ePSFs 
+    simultaneously.
+
+    Returns
+    -------
+
+    Downloads the ePSF FITS files to download_location. The explicit return is 
+    a list of the downloaded FITS files.
+
+    """
+
+    # select if either choosing to provide inputs via a text file or list
+    if fromTextFile == True:
+        with open(input_list) as file:
+            lines = [line.rstrip() for line in file]
```

Comment:
This is something @pllim may have more to say about, but I think you'll need some smart handling/errors for poorly-formatted input files.